src/app/content/highlights/reducer.ts

```diff
@@ -76,6 +78,10 @@ const reducer: Reducer<State, AnyAction> = (state = initialState, action) => {
         return state;
       }
 
+      const hasUnsavedHighlight =
+        oldHighlight.annotation === action.payload.highlight.annotation
+        && state.hasUnsavedHighlight;
```

Comment:
Value for `state.hasUnsavedHighlight` is determined by current value `state.hasUnsavedHighlight`. It is misleading for me, why is that?

Reply:
Well it was `onlyColorHasChanged` before. This action triggers when either note is saved or color is changed hence why it needs to carry over. When color is changed, the action does not have the new annotation.